src/sql/SqlResult.ts

```diff
@@ -319,6 +335,9 @@ export class SqlResultImpl implements SqlResult {
     /** Returns if there are rows to be iterated. Used by {@link next}. */
     hasNext(): Promise<boolean> {
         return this.executeDeferred.promise.then(() => {
+            return this.isRowSet();
```

Comment:
We shouldn't need this extra promise chain (`return this.isRowSet();) We could check it already in the `then` block below by looking at the updateCount

Reply:
yeah good point